### PR TITLE
refactor(openapi/rust): use paperclip as code generator

### PR DESCRIPTION
**Pull request by tiagolobocastro**
We can now stop using our fork and use paperclip which allows us to keep the templates locally on openapi/templates.

### CODE REVIEW

**Review by niladrih — APPROVED** [review]

File: nix/pkgs/control-plane/cargo-project.nix [hunk 1]
```diff
@@ -71,8 +72,8 @@ let
     GIT_VERSION = "${gitVersions.tag_or_long}";
 
     inherit LIBCLANG_PATH PROTOC PROTOC_INCLUDE;
-    nativeBuildInputs = [ clang pkg-config openapi-generator which git llvmPackages.bintools ];
-    buildInputs = [ llvmPackages.libclang protobuf openssl.dev systemdMinimal.dev utillinux.dev rdma-core ];
+    nativeBuildInputs = [ clang pkg-config paperclip which git llvmPackages.bintools ];
+    buildInputs = [ llvmPackages.libclang protobuf openssl.dev systemdMinimal.dev utillinux.dev ];
```
Comment (dsharma-dc) [inline nix/pkgs/control-plane/cargo-project.nix]: `rdma-core` removal is intentional here? I didn't think that's redundant here.

**Review by Abhinandan-Purkait — APPROVED** [review]
Looks good. Python BDD test code generation will happen through openapi-generator-cli?

File: nix/pkgs/control-plane/cargo-project.nix (hunk 1 above)
Comment (tiagolobocastro) [reply nix/pkgs/control-plane/cargo-project.nix]: oops, good catch, it was bad rebase, fixed!

**Comment by tiagolobocastro** [comment]
> Looks good. Python BDD test code generation will happen through openapi-generator-cli?

Yeah that will still use it, but the upstream one rather than our fork.

File: openapi/templates/default/README.mustache [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+To get access to the crate's generated documentation, use:
+
+```
+cargo doc --open
+```
+
+## Author
+
+{{#apiInfo}}{{#apis}}{{#-last}}{{{infoEmail}}}
+{{/-last}}{{/apis}}{{/apiInfo}}
```
Comment (dsharma-dc) [inline openapi/templates/default/README.mustache:50]: nit: blank line needed? shows as 🚫 in many files

Comment (tiagolobocastro) [reply openapi/templates/default/README.mustache:50]: Just means there's no newline at the end

**Comment by tiagolobocastro** [comment]
bors merge

File: shell.nix [hunk 3]
```diff
@@ -33,14 +33,15 @@ mkShell {
     jq
     llvmPackages.libclang
     nixpkgs-fmt
-    openapi-generator
+    openapi-generator-cli
```
Comment (dsharma-dc) [inline shell.nix:36]: Since we are not going to use `openapi-generator-cli`, better remove from here?

**Review by dsharma-dc — APPROVED** [review]

File: shell.nix (hunk 3 above)
Comment (dsharma-dc) [reply shell.nix:36]: ok, from another comment I understand it'll still be used for python bdd tests.

Comment (tiagolobocastro) [reply shell.nix:36]: Yes, still need it, but now it's from upstream and cached by hydra

**Comment by tiagolobocastro** [comment]
bors merge

**Review by sinhaashish — APPROVED** [review]

**Comment by tiagolobocastro** [comment]
bors merge

**Pull request merged**